pkgs/tools/misc/trackma/default.nix

```
@@ -6,10 +6,24 @@
 , gobject-introspection
 , glib
 , gtk3
+, qt5
+, makeDesktopItem
 , withCurses ? false
 , withGtk ? false
+, withQt ? false
 }:
-
+let
+  mkDesktopItem = pname: dname: pcomment: isTerminal: makeDesktopItem rec {
+    name = pname;
+    icon = "trackma";
+    exec = name + " %u";
+    desktopName = dname;
+    type = "Application";
+    categories = [ "Network" ];
+    comment = pcomment;
+    terminal = isTerminal;
```

Your code lacks elegance.

You are using `param = Xparam;`, and this is not a good practice in Nix code in general and Nixpkgs in particular. 
Usually you should use the same parameter name and `inherit` it.

```suggestion
  mkDesktopItem = name: desktopName: comment: terminal: makeDesktopItem {
    inherit name desktopName comment terminal;
    icon = "trackma";
    exec = name + " %u";
    type = "Application";
    categories = [ "Network" ];
```

(AFAIR the `rec` keyword is superfluous here.)